up42/utils.py

```diff
@@ -300,6 +320,119 @@ def _plot_images(
+    """
+
+    centroid = box(*result_df.total_bounds).centroid
+    m = folium_base_map(
+        lat=centroid.y,
+        lon=centroid.x,
+    )
+
+    df_bounds = result_df.bounds
+    list_bounds = df_bounds.values.tolist()
+    raster_filepaths = [
+        path for path in filepaths if Path(path).suffix in plot_file_format
+    ]
+
+    # Make sure the quicklooks images are equal to the number of features in the scenes
+    try:
+        assert len(list_bounds) == len(raster_filepaths)
+    except AssertionError:
+        logger.error(
+            f"The length of the imgaes {len(raster_filepaths)} is not equal to number of"
+            f"row in scenes {len(result_df)}"
+        )
+        raise
+
+    try:
+        feature_names = result_df[name_column].to_list()
+    except KeyError:
+        feature_names = [""] * len(result_df.index)
+
+    if aoi is not None:
+        folium.GeoJson(
+            aoi,
+            name="geojson",
+            style_function=_style_function,
+            highlight_function=_highlight_function,
+        ).add_to(m)
+
+    if show_features:
+        for idx, row in result_df.iterrows():  # type: ignore
+            try:
+                feature_name = row.loc[name_column]
+            except KeyError:
+                feature_name = ""
+            layer_name = f"Feature {idx+1} - {feature_name}"
+            f = folium.GeoJson(
+                row["geometry"],
+                name=layer_name,
+                style_function=_style_function,
+                highlight_function=_highlight_function,
+            )
+            folium.Popup(
+                f"{layer_name}: {row.drop('geometry', axis=0).to_json()}"
+            ).add_to(f)
+            f.add_to(m)
+
+    if show_images and raster_filepaths:
+        for idx, (raster_fp, feature_name) in enumerate(
+            zip(raster_filepaths, feature_names)
+        ):
+            with rasterio.open(raster_fp) as src:
+                if src.meta["crs"] is None:
+                    dst_array = src.read()[:3, :, :]
+                    minx, miny, maxx, maxy = list_bounds[idx]
+                else:
+                    # Folium requires 4326, streaming blocks are 3857
+                    with WarpedVRT(src, crs="EPSG:4326") as vrt:
+                        dst_array = vrt.read()[:3, :, :]
+                        minx, miny, maxx, maxy = vrt.bounds
+
+            m.add_child(
+                folium.raster_layers.ImageOverlay(
+                    np.moveaxis(np.stack(dst_array), 0, 2),
+                    bounds=[[miny, minx], [maxy, maxx]],  # different order.
+                    name=f"Image {idx + 1} - {feature_name}",
+                )
+            )
+
+    # Collapse layer control with too many features.
+    collapsed = bool(result_df.shape[0] > 4)
+    folium.LayerControl(position="bottomleft", collapsed=collapsed).add_to(m)
+
+    if save_html:
+        if not Path(save_html).exists():
```

Comment:
`save_html = Path(save_html)` once at the top probably best.

Reply:
Ok, I will do